src/tfvc/uihelper.ts

```diff
@@ -0,0 +1,57 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+"use strict";
+
+import { QuickPickItem, window, workspace } from "vscode";
+import { IPendingChange } from "./interfaces";
+
+var path = require("path");
+
+export class UIHelper {
+    public static async ChoosePendingChange(changes: IPendingChange[]): Promise<IPendingChange> {
+        if (changes && changes.length > 0) {
+            // First, create an array of quick pick items from the changes
+            let items: QuickPickItem[] = [];
+            for (let i = 0; i < changes.length; i++) {
+                items.push({
+                    label: UIHelper.GetFileName(changes[i]),
+                    description: changes[i].changeType,
+                    detail: UIHelper.GetRelativePath(changes[i])
+                    });
+            }
+            // Then, show the quick pick window and get back the one they chose
+            //TODO localize strings
+            let item: QuickPickItem = await window.showQuickPick(
+                items, { matchOnDescription: true, placeHolder: "Choose a file to open the file." });
```

Comment:
Agreed. Just trying to let the design settle a bit before doing that. Let's talk about how it should be done on Monday.